### ISSUE TITLE
Same generations When GRPO with Qwen2.5VL

### ISSUE THREAD

**Issue by aaaalonga** [modelscope/ms-swift#5967]
Here ara my Parameters
    --num_generations 8 \
    --temperature 1.2 \
    --top_k 100 \
    --top_p 0.95 \
but I found the generations are same whatever I used Qwen2.5VL or finetuned Qwen2.5VL
Did I miss some parameters to set up?

**Comment by hjh0119** [comment]
please provide the environment infos

**Comment by aaaalonga** [comment]
Here is my environment：
absl-py                       2.3.1
accelerate                    1.8.1
addict                        2.4.0
aiofiles                      24.1.0
aiohappyeyeballs              2.6.1
aiohttp                       3.12.13
aiosignal                     1.4.0
aliyun-python-sdk-core        2.16.0
aliyun-python-sdk-kms         2.16.5
annotated-types               0.7.0
antlr4-python3-runtime        4.13.2
anyio                         4.9.0
asttokens                     3.0.0
async-timeout                 5.0.1
attrdict                      2.0.1
attrs                         25.3.0
av                            15.0.0
binpacking                    1.5.2
blis                          1.3.0
catalogue                     2.0.10
certifi                       2025.7.9
cffi                          1.17.1
charset-normalizer            3.4.2
click                         8.2.1
cloudpathlib                  0.22.0
comm                          0.2.3
confection                    0.1.5
contourpy                     1.3.2
cpm-kernels                   1.0.11
crcmod                        1.7
cryptography                  45.0.5
cycler                        0.12.1
cymem                         2.0.11
dacite                        1.9.2
datasets                      3.3.2
debugpy                       1.8.16
decorator                     5.2.1
decord                        0.6.0
deepspeed                     0.17.2
dill                          0.3.8
distro                        1.9.0
einops                        0.8.1
en_core_web_sm                3.8.0
et_xmlfile                    2.0.0
exceptiongroup                1.3.0
executing                     2.2.0
fastapi                       0.116.0
ffmpy                         0.6.0
filelock                      3.18.0
fonttools                     4.58.5
frozenlist                    1.7.0
fsspec                        2024.12.0
future                        1.0.0
gradio                        5.35.0
gradio_client                 1.10.4
groovy                        0.1.2
grpcio                        1.73.1
h11                           0.16.0
hf-xet                        1.1.5
hjson                         3.1.0
httpcore                      1.0.9
httpx                         0.28.1
huggingface-hub               0.33.2
idna                          3.10
imageio                       2.37.0
importlib_metadata            8.7.0
ipykernel                     6.30.1
ipython                       8.37.0
jedi                          0.19.2
jieba                         0.42.1
Jinja2                        3.1.6
jiter                         0.10.0
jmespath                      0.10.0
joblib                        1.5.1
json_repair                   0.51.0
jupyter_client                8.6.3
jupyter_core                  5.8.1
kiwisolver                    1.4.8
langcodes                     3.5.0
language_data                 1.3.0
latex2sympy2_extended         1.0.6
lazy_loader                   0.4
marisa-trie                   1.3.1
Markdown                      3.8.2
markdown-it-py                3.0.0
MarkupSafe                    3.0.2
math-verify                   0.5.2
matplotlib                    3.10.3
matplotlib-inline             0.1.7
mdurl                         0.1.2
modelscope                    1.28.0
mpmath                        1.3.0
ms_swift                      3.9.0.dev0  ./ms-swift
msgpack                       1.1.1
multidict                     6.6.3
multiprocess                  0.70.16
murmurhash                    1.0.13
nest_asyncio                  1.6.0
networkx                      3.4.2
ninja                         1.11.1.4
nltk                          3.9.1
numpy                         1.26.0
nvidia-cublas-cu12            12.6.4.1
nvidia-cuda-cupti-cu12        12.6.80
nvidia-cuda-nvrtc-cu12        12.6.77
nvidia-cuda-runtime-cu12      12.6.77
nvidia-cudnn-cu12             9.5.1.17
nvidia-cufft-cu12             11.3.0.4
nvidia-cufile-cu12            1.11.1.6
nvidia-curand-cu12            10.3.7.77
nvidia-cusolver-cu12          11.7.1.2
nvidia-cusparse-cu12          12.5.4.2
nvidia-cusparselt-cu12        0.6.3
nvidia-nccl-cu12              2.26.2
nvidia-nvjitlink-cu12         12.6.85
nvidia-nvtx-cu12              12.6.77
openai                        1.93.2
opencv-python                 4.9.0.80
openpyxl                      3.1.5
orjson                        3.10.18
oss2                          2.19.1
packaging                     25.0
pandas                        2.3.1
parso                         0.8.4
peft                          0.11.0
pexpect                       4.9.0
pickleshare                   0.7.5
pillow                        11.3.0
pip                           25.1.1
platformdirs                  4.3.8
preshed                       3.0.10
prompt_toolkit                3.0.51
propcache                     0.3.2
protobuf                      6.31.1
psutil                        7.0.0
ptyprocess                    0.7.0
pure_eval                     0.2.3
py-cpuinfo                    9.0.0
pyarrow                       20.0.0
pycocotools                   2.0.10
pycparser                     2.22
pycryptodome                  3.23.0
pydantic                      2.11.7
pydantic_core                 2.33.2
pydub                         0.25.1
Pygments                      2.19.2
pyparsing                     3.2.3
python-dateutil               2.9.0.post0
python-multipart              0.0.20
pytz                          2025.2
PyYAML                        6.0.2
pyzmq                         27.0.1
qwen-vl-utils                 0.0.11
regex                         2024.11.6
requests                      2.32.4
rich                          14.0.0
rouge                         1.0.1
ruff                          0.12.2
safehttpx                     0.1.6
safetensors                   0.5.3
scikit-image                  0.25.2
scipy                         1.15.3
semantic-version              2.10.0
sentencepiece                 0.2.0
setuptools                    80.9.0
shellingham                   1.5.4
simplejson                    3.20.1
six                           1.17.0
smart_open                    7.3.1
sniffio                       1.3.1
sortedcontainers              2.4.0
spacy                         3.8.7
spacy-legacy                  3.0.12
spacy-loggers                 1.0.5
srsly                         2.5.1
stack_data                    0.6.3
starlette                     0.46.2
sympy                         1.14.0
tensorboard                   2.19.0
tensorboard-data-server       0.7.2
thinc                         8.3.6
tifffile                      2025.5.10
tiktoken                      0.9.0
tokenizers                    0.21.2
tomlkit                       0.13.3
torch                         2.7.1
torchvision                   0.22.1
tornado                       6.5.2
tqdm                          4.67.1
traitlets                     5.14.3
transformers                  4.52.3
transformers-stream-generator 0.0.5
triton                        3.3.1
trl                           0.19.1
typer                         0.16.0
typing_extensions             4.14.1
typing-inspection             0.4.1
tzdata                        2025.2
urllib3                       2.5.0
uvicorn                       0.35.0
wasabi                        1.1.3
wcwidth                       0.2.13
weasel                        0.4.1
websockets                    15.0.1
Werkzeug                      3.1.3
wheel                         0.45.1
wrapt                         1.17.3
xxhash                        3.5.0
yarl                          1.20.1
zipp                          3.23.0
zstandard                     0.23.0

Here is my complete script

MAX_PIXELS=${MAX_PIXELS} \
MIN_PIXELS=${MIN_PIXELS} \
MASTER_PORT=29100 \
NPROC_PER_NODE=$nproc_per_node \
CUDA_VISIBLE_DEVICES=0,1 \
swift rlhf \
    --rlhf_type grpo \
    --model ./merged_model/checkpoint \
    --model_type qwen2_5_vl \
    --external_plugins ./plugin/plugin_gmner.py \
    --reward_funcs my_reward \
    --custom_register_path  ./custom/dataset_grpo.py \
    --dataset 'grpo' \
    --output_dir output_dir \
    --split_dataset_ratio 0 \
    --train_type lora \
    --lora_rank 8 \
    --lora_alpha 32 \
    --target_modules all-linear \
    --freeze_vit False \
    --freeze_aligner False \
    --torch_dtype bfloat16 \
    --max_completion_length 1024 \
    --num_train_epochs 2 \
    --per_device_train_batch_size 4 \
    --per_device_eval_batch_size 4 \
    --learning_rate 5e-7 \
    --gradient_accumulation_steps $(expr 16 / $nproc_per_node) \
    --eval_steps 50 \
    --save_steps 250 \
    --save_total_limit 15 \
    --logging_steps 50 \
    --max_length 1024 \
    --warmup_ratio 0.05 \
    --dataloader_num_workers 4 \
    --dataset_num_proc 4 \
    --num_generations 8 \
    --temperature 1.2 \
    --top_k 100 \
    --top_p 0.95 \
    --log_completions true \
    --num_iterations 1 \
    --beta 0.001

**Comment by hjh0119** [comment]
I tested with the script below (replacing model, dataset, and reward), and it runs correctly on my side.


```bash
MAX_PIXELS=602112 \
MASTER_PORT=29100 \
NPROC_PER_NODE=2 \
CUDA_VISIBLE_DEVICES=0,1 \
swift rlhf \
    --rlhf_type grpo \
    --model Qwen/Qwen2.5-VL-3B-Instruct \
    --model_type qwen2_5_vl \
    --external_plugins examples/train/grpo/plugin/plugin.py \
    --reward_funcs external_r1v_acc format \
    --dataset 'AI-ModelScope/clevr_cogen_a_train' \
    --split_dataset_ratio 0 \
    --train_type lora \
    --lora_rank 8 \
    --lora_alpha 32 \
    --target_modules all-linear \
    --freeze_vit False \
    --freeze_aligner False \
    --torch_dtype bfloat16 \
    --max_completion_length 1024 \
    --num_train_epochs 2 \
    --per_device_train_batch_size 4 \
    --per_device_eval_batch_size 4 \
    --learning_rate 5e-7 \
    --gradient_accumulation_steps 8 \
    --eval_steps 50 \
    --save_steps 250 \
    --save_total_limit 15 \
    --logging_steps 50 \
    --max_length 1024 \
    --warmup_ratio 0.05 \
    --dataloader_num_workers 4 \
    --dataset_num_proc 4 \
    --num_generations 8 \
    --temperature 1.2 \
    --top_k 100 \
    --top_p 0.95 \
    --log_completions true \
    --num_iterations 1 \
    --beta 0.001
```

**Comment by hjh0119** [comment]
> transformers 4.52.3

maybe you could upgrade transformers and retry
